### PR TITLE
Adding support for getting env config through client env

### PR DESCRIPTION
**Pull request by bandana147**
* Make IMS environment configurable from config in standalone navigation

Resolves: [MWPW-157474](https://jira.corp.adobe.com/browse/MWPW-157474)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://MWPW-157474--milo--adobecom.hlx.page/?martech=off

**QA**
- https://adobecom.github.io/nav-consumer/navigation.html?navbranch=mwpw-157474

### CODE REVIEW

**Comment by narcis-radu** [comment]
@bandana147 - what's the use-case for this enhancement?

**Comment by bandana147** [comment]
> @bandana147 - what's the use-case for this enhancement?

This applies to standalone gnav clients. If their page is not listed in this condition:

`if (host.includes(`${SLD}.page`)
    || host.includes(`${SLD}.live`)
    || host.includes('stage.adobe')
    || host.includes('corp.adobe')
    || host.includes('graybox.adobe')) {
    return { ...ENVS.stage, consumer: conf.stage };
}`
it always defaults to production.
We did suggest using the env URL parameter, but it seems to get lost sometimes during redirection. 
They’ve requested the ability to pass the env parameter while configuring the navigation, to ensure they can consistently use the correct environment for their stage setup.

File: libs/utils/utils.js [hunk 1]
```diff
@@ -150,8 +150,11 @@ const PROMO_PARAM = 'promo';
 function getEnv(conf) {
   const { host } = window.location;
   const query = PAGE_URL.searchParams.get('env');
-
   if (query) return { ...ENVS[query], consumer: conf[query] };
+
+  const { clientEnv } = conf;
```
Comment (mokimo) [inline libs/utils/utils.js:155]: I wonder if we should more generically look for `window.miloClientEnv` ? 
That would work for blocks exported outside of the gnav as well

**Review by mokimo — APPROVED** [review]
new config, new unit test!

File: libs/utils/utils.js (hunk 1 above)
Comment (bandana147) [reply libs/utils/utils.js:155]: Yes, I  will update it to `window.miloClientEnv`

**Review by sonawanesnehal3 — APPROVED** [review]

**Review by sharmrj — APPROVED** [review]

**Review by salonijain3 — APPROVED** [review]

**Comment by prativas22** [comment]
The PR is validated, and the test results are updated in the ticket's subtask.
https://jira.corp.adobe.com/browse/MWPW-157474

https://adobecom.github.io/nav-consumer/navigation.html?navbranch=mwpw-157474&authoringpath=/federal/dev/gnavtest/ccgnav
image
image

File: libs/navigation/navigation.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ export default async function loadBlock(configs, customLib) {
   const {
     header, footer, authoringPath, env = 'prod', locale = '', theme,
   } = configs || {};
+  window.miloClientEnv = env;
```
Comment (narcis-radu) [inline libs/navigation/navigation.js:40]: we should not export a global variable with environment details

File: libs/utils/utils.js (hunk 1 above)
Comment (bandana147) [reply libs/utils/utils.js]: @narcis-radu Initially it was added as part of config. Should we keep it this way?
cc: @salonijain3

**Pull request merged**